src/esp/physics/objectManagers/PhysicsObjectBaseManager.h

```diff
@@ -0,0 +1,158 @@
+
+  /**
+   * @brief set the weak reference to the physics manager that owns this wrapper
+   * manager
+   */
+  void setPhysicsManager(
+      const std::shared_ptr<esp::physics::PhysicsManager>& physMgr) {
+    weakPhysManager_ = physMgr;
+  }
+
+  /**
+   * @brief Creates an empty @ref esp::physics::AbstractManagedPhysicsObject of
+   * the type managed by this manager.
+   *
+   * @param objectHandle Unused.  Object being wrapped will provide its name.
+   * @param registerObject whether to add this managed object to the
+   * library or not. If the user is going to edit this managed object, this
+   * should be false. Defaults to true. If specified as true, then this function
+   * returns a copy of the registered managed object.
+   * @return a reference to the desired managed object.
+   */
+  ObjWrapperPtr createObject(
+      const std::string& objectHandle,
+      CORRADE_UNUSED bool registerObject = false) override;
+
+ protected:
+  /**
+   * @brief Any physics-object-wrapper-specific resetting that needs to happen
+   * on reset.
+   */
+  void resetFinalize() override {}
+
+  /**
+   * @brief This method will remove objects from physics manager.  The wrapper
+   * has already been removed by the time this method is called (this is called
+   * from @ref esp::core::ManagedContainerBase::deleteObjectInternal)
+   *
+   * @param objectID the ID of the managed object to remove
+   * @param objectHandle the string key of the managed object to remove.
+   */
+  void deleteObjectInternalFinalize(
+      int objectID,
+      CORRADE_UNUSED const std::string& objectHandle) override {
+    if (auto physMgr = this->getPhysicsManager()) {
+      if (physMgr->isValidObjectID(objectID)) {
+        physMgr->removeObject(objectID);
+      }
+    }
+  }  // updateObjectHandleLists
+
+  /**
+   * @brief implementation of managed object type-specific registration
+   * @param object the managed object to be registered
+   * @param objectHandle the name to register the managed object with.
+   * Expected to be valid.
+   * @param forceRegistration Will register object even if conditional
+   * registration checks fail.
+   * @return The unique ID of the managed object being registered, or
+   * ID_UNDEFINED if failed
+   */
+  int registerObjectFinalize(ObjWrapperPtr object,
+                             const std::string& objectHandle,
+                             CORRADE_UNUSED bool forceRegistration) override {
+    // Add wrapper to template library
+    return this->addObjectToLibrary(object, objectHandle);
+  }  // PhysicsObjectBaseManager::registerObjectFinalize
+
+  /**
+   * @brief return a reference to physicsManager_, or null ptr if it does not
+   * exist anymore.  This is necessary since a reference of this manager may
+   * linger in python after simulator/physicsManager get clobbered/rebuilt.
+   */
+  std::shared_ptr<esp::physics::PhysicsManager> getPhysicsManager() const {
+    auto sp = weakPhysManager_.lock();
+    if (!sp) {
```

Comment:
Let's make this an ESP_CHECK that way we can throw an error.  It's good that keeping an object after the simulator is deleted won't cause a segfault, but I can't think of a reason why we shouldn't loudly inform the user that this object reference is  no longer controlling anything.  Can you also also expose an `isAlive()` method (or similar) that checks to make sure the object wrapper's physic manager is still alive and the ID still exists in the physics manager (if it's alive).  That way someone can easily do

```python
my_objs = list(filter(lambda obj: not obj.is_alive, my_objs))
```
to get rid of any dead objects?

Reply:
Great idea re:isAlive(), thanks!

Reply:
(Apologies if my comment is totally out of place, I don't know much about this area of Habitat's code.) Isn't it possible to ensure the originating object stays around, instead of emitting an error or managing "dead objects"? I'm doing this for a lot of Magnum objects already (array views keep the originating memory owner around, meshes keep their buffers around etc.).

Reply:
@mosra I think the main issue is memory reload/restarts, right? Also you could theoretically hotswap managers so those errors would be needed if I understand correctly, right?